### PR TITLE
Add optional parameter ambassadorNodePort. 

### PR DESCRIPTION
**Pull request by iankoulski**
Useful when ambassadorServiceType is NodePort. Allows fixing the port number to a specific value which can be used when configuring a load balancer or another application that connects to this port. This value should be set to a number between 30000 and 32767. By default it is set to 30080. If not set and ambassadorServiceType is NodePort, a random available port value within the nodePort range will be assigned.

Contributes to closing the following issues:
https://github.com/kubeflow/kubeflow/issues/630
https://github.com/kubeflow/kubeflow/issues/1525

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2457)

### CODE REVIEW

**Comment by iankoulski** [comment]
CLA Signed!

**Comment by iankoulski** [comment]
Added alternate email address to GitHub

**Comment by iankoulski** [comment]
CLA Signed

File: kubeflow/common/prototypes/ambassador.jsonnet [hunk 1]
```diff
@@ -5,6 +5,7 @@
 // @param name string Name
 // @optionalParam platform string none supported platforms {none|gke|minikube}
 // @optionalParam ambassadorServiceType string ClusterIP The service type for the API Gateway.
+// @optionalParam ambassadorNodePort number 30080 The port number to expose ambassador on when ambassadorServiceType is NodePort
```
Comment (kunmingg) [inline kubeflow/common/prototypes/ambassador.jsonnet]: Thanks for the PR!
Can you change default value of ambassadorNodePort to null and only add
```
nodePort: params.ambassadorNodePort
```
When ambassadorServiceType is NodePort and ambassadorNodePort is not null?

File: kubeflow/common/ambassador.libsonnet [hunk 2]
```diff
@@ -20,6 +20,10 @@
             name: "ambassador",
             port: 80,
             targetPort: 80,
+            [if (params.ambassadorServiceType == 'NodePort') && 
+                (params.ambassadorNodePort >= 30000) && 
+                (params.ambassadorNodePort <= 32767) 
+             then 'nodePort']: params.ambassadorNodePort,
```
Comment (kunmingg) [inline kubeflow/common/ambassador.libsonnet]: ```: params.ambassadorNodePort,```should be within if condition as well?

File: kubeflow/common/prototypes/ambassador.jsonnet (hunk 1 above)
Comment (iankoulski) [reply kubeflow/common/prototypes/ambassador.jsonnet]: Hi @kunmingg, I've implemented the change with the conditions described above. I kept the ambassadorNodePort parameter as a number with default value 0. If the ambassadorServiceType is changed to NodePort, by default, a random available port will be picked by k8s. The nodePort will be included in the configuration only if the parameter is set to a valid number from the range.

File: kubeflow/common/ambassador.libsonnet (hunk 2 above)
Comment (iankoulski) [reply kubeflow/common/ambassador.libsonnet]: Apparently If the the if statement evaluates to false, libsonnet will not include the rest of the line in the yaml. I left the code this way and tested to make sure it behaves properly in the different scenarios.

**Comment by iankoulski** [comment]
In order to correct the issue with my CLA and also catch up with the changes in the master branch, I had to create a new pull request: https://github.com/kubeflow/kubeflow/pull/2511
Closing this one, please review the new pull request instead.
Thank You!

**Pull request closed without merging**